lms/services/blackboard_api/service.py

```diff
@@ -53,12 +53,38 @@ def list_files(self, course_id):
         :raise ProxyAPIAccessTokenError: if we don't have a Blackboard API
             access token for the current user
         """
-        return self._http_service.get(
+        files = self._http_service.get(
             self._api_url(f"courses/uuid:{course_id}/resources"),
             oauth=True,
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+        return [
+            {
+                "id": self._blackboard_document_url(file_["id"]),
+                "display_name": file_["name"],
+                "updated_at": file_["modified"],
+            }
+            for file_ in files
+        ]
+
+    def public_url(self, course_id, file_id):
+        files = self._http_service.get(
+            self._api_url(f"/courses/uuid:{course_id}/resources"),
+            oauth=True,
+            schema=BlackboardListFilesSchema,
+        ).validated_data
+
+        for file_ in files:
+            if self._blackboard_document_url(file_["id"]) == file_id:
```

Comment:
I know the ship has sailed on the whole `blackboard://...` but it seems like a lot of self inflicted complexity.

Reply:
We haven't actually released anything yet, so it's not too late to change our minds. Why do you think it's complex? What would you suggest instead?

Reply:
I think `blackboard://content-resource/{file_id}/` is more complex than just `file_id` I can really find any justification to it other than "the field is the DB is called document_url" and this `blackboard://` makes it, technically, an URL.

There's another PR open at the moment to persist canvas Files, I think we could reevaluate how we store this id once we integrate width that (I don't think there's an immediate need to store BB files but we probably should do it anyway mid-term)